### DISCUSSION TITLE
Introduce `private` (or equivalent) keyword for functions and vars

### DISCUSSION THREAD

**Discussion by sa-** [modular/modular#129] [Ideas]
This should be backwards compatible with python, the convention of using a leading underscore still allows one to access the function/members from outside the class.

Interested to hear what the community has to say

**Comment by lattner** [comment]
This is a very clear extension we could consider, highly precedented of course.  In the immediate future we are focusing on building the core systems programming features in the roadmap.  When that is complete, we can consider "general goodness" features like this.
